Firestore/core/src/firebase/firestore/nanopb/nanopb_string.h

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2018 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_NANOPB_NANOPB_STRING_H_
+#define FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_NANOPB_NANOPB_STRING_H_
+
+#include <pb.h>
+
+#include <cstdlib>
+#include <functional>
+#include <string>
+#include <utility>
+
+#include "Firestore/core/src/firebase/firestore/util/comparison.h"
+#include "absl/strings/string_view.h"
+
+namespace firebase {
+namespace firestore {
+namespace nanopb {
+
+class String : public util::Comparable<String> {
+ public:
+  static pb_bytes_array_t* MakeBytesArray(absl::string_view value) {
+    auto size = static_cast<pb_size_t>(value.size());
```

Comment:
This won't work if string_view contains `wchar_t`'s rather than `char`'s. Fortunately, absl::string_view only supports chars, so we should be ok.

But if you want to future proof us for std::string_view (which does support wchar_t's) then you could either:
a) Add an assert that checks the size/type of string_view::value_type.
b) Actually perform the math. (Dubious. The extra complication seems unwarranted just now since it can't possibly be useful until we absl::string_view -> std::string_view.)